docker/docker_image_src.go

```diff
@@ -83,6 +83,26 @@ func (s *dockerImageSource) GetManifest() ([]byte, string, error) {
 	return s.cachedManifest, s.cachedManifestMIMEType, nil
 }
 
+// GetTargetManifest ...
+func (s *dockerImageSource) GetTargetManifest(reference string) ([]byte, string, error) {
```

Comment:
This will need a documented semantics for `reference` (this is definitely not a Docker reference, nor a `ImageReference`. I guess the input should be a digest, of perhaps a `types.BlobInfo`, like in `GetBlob`.

> OCI has this concept of a manifest list also, it makes sense have this new method in the image source interface. `GetManifestFromList(arch, platform, ...)` or `GetTargetManifest(arch, platform, ...)`.

I think parsing the lists and matching by arch/platform belongs to the manifest parsing code in `image`, not in the individual `ImageSource`s.